pkg/resourceinterpreter/defaultinterpreter/healthy_test.go

```diff
@@ -211,3 +212,314 @@ func Test_interpretStatefulSetHealth(t *testing.T) {
 		})
 	}
 }
+
+func Test_interpretReplicaSetHealth(t *testing.T) {
+	tests := []struct {
+		name    string
+		object  *unstructured.Unstructured
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "failed convert to ReplicaSet object",
+			object: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"spec":   "format error",
+					"status": "format error",
+				},
+			},
+			want:    false,
+			wantErr: true,
+		},
```

Comment:
I see similar judgments in each test case. Is it possible to consider testing method `ConvertToTypedObject` so that similar test cases can be omitted?

If this is feasible, you can submit a separate pr to test method `ConvertToTypedObject`.